File: tests/journal.lib/debug_empty.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2020 all rights reserved


// get the journal
#include <p2/journal.h>
// support
#include <cassert>


// verify that empty messages are handled correctly
int main() {
    // make an info channel
    pyre::journal::debug_t channel("tests.journal.debug");
    // send the output to the trash
    channel.device(std::make_shared<pyre::journal::trash_t>());

    // inject nothing
    channel << pyre::journal::endl;

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/debug_empty.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2020 all rights reserved


def test():
    """
    Verify that empty log messages get handled properly
    """
    # access the journal
    import j2

    # make a debug channel
    channel = j2.debug(name="tests.journal.debug")
    # activate it
    channel.activate()
    # but send the output to trash
    channel.device = j2.trash()

    # inject
    channel.log()

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
